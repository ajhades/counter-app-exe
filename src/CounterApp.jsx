import { useState } from 'react';
import PropTypes from 'prop-types';

export const CounterApp = ({ value }) => {
  const [counter, setCounter] = useState(value);

  const handleAdd = () => {
    setCounter(counter + 1);
  }
  const handleSub = () => setCounter(counter - 1);
  const handleReset = () => setCounter(value);
  return (
    <>
      <h1> CounterApp </h1>
      <h1> {counter} </h1>

      <button onClick={ handleAdd }>Increment +1 </button>
      <button onClick={ handleSub }>Substract -1 </button>
      <button onClick={ handleReset }>Reset counter </button>
    </>
  );
};

CounterApp.propTypes = {
  value: PropTypes.number.isRequired,
}