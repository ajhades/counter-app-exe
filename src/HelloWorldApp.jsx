import PropTypes from 'prop-types';

export const HelloWorldApp = ({title, subtitle}) => {
  return (
    <>
      <h1>{ title }</h1>
      <p>{ subtitle }</p>
    </>
  );
};

HelloWorldApp.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string.isRequired,
}

HelloWorldApp.defaultProps = {
  title: 'Hello, World!',
  subtitle: 'Welcome to this wonderful app!',
}