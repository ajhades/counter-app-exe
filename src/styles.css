html,body{
  background-color: darkslategrey;
  color:white;
  font-family: Arial, Helvetica, sans-serif;
  font-size: 1.3rem;
}