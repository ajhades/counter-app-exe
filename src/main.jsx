import React from "react";
import ReactDOM from "react-dom/client";
// import { HelloWorldApp } from "./HelloWorldApp";
import './styles.css';
import { CounterApp } from "./CounterApp";

ReactDOM.createRoot( document.getElementById('root')).render(
  <React.StrictMode>
    <CounterApp value={12}/>
  </React.StrictMode>
);

