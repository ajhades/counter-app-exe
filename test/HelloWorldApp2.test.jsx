import { render, screen } from "@testing-library/react";
import { HelloWorldApp } from "../src/HelloWorldApp";

describe("Prueba en <HelloWorldApp>", () => {
  const title = "Hello World";
  const subtitle = "World of the world";

  test("debe de hacer match con el snapshot", () => {
    const { container } = render(
      <HelloWorldApp title={title} subtitle={subtitle} />
    );

    expect(container).toMatchSnapshot();
  });

  test("debe motrar el titulo Hello World", () => {
    render(<HelloWorldApp title={title} subtitle={subtitle} />);

    expect(screen.getByText(title)).toBeTruthy();
  });

  test("debe motrar el titulo en h1", () => {
    render(<HelloWorldApp title={title} subtitle={subtitle} />);

    expect(screen.getByRole("heading", { level: 1 }).innerHTML).toContain(
      title
    );
  });

  test("debe motrar el subtitulo enviado en props", () => {
    render(<HelloWorldApp title={title} subtitle={subtitle} />);

    expect(screen.getByText(subtitle)).toBeTruthy();
  });
});
