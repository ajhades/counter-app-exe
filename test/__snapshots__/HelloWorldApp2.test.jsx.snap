// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en <HelloWorldApp> debe de hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hello World
  </h1>
  <p>
    World of the world
  </p>
</div>
`;
